src/compressor/QatAccel.cc

```diff
@@ -11,19 +11,27 @@
  * Foundation.  See file COPYING.
  *
  */
-
+#include <qatzip.h>
 #include "QatAccel.h"
 
+void QzSessionDeleter::operator() (struct QzSession_S *session) {
+  if (NULL != session->internal) {
+    qzTeardownSession(session);
+    qzClose(session);
+  }
+  delete session;
+}
+
 /* Estimate data expansion after decompression */
 static const unsigned int expansion_ratio[] = {5, 20, 50, 100, 200};
 
-QatAccel::~QatAccel() {
-  if (NULL != session.internal) {
-    qzTeardownSession(&session);
-    qzClose(&session);
-  }
+QatAccel::QatAccel() {
+  session.reset(new struct QzSession_S);
```

Comment:
could use `make_unique<>` if you please.

Reply:
@tchaikov

Reply:
@tchaikov do you have any other idea, or is that ok to use `new`?